### DISCUSSION TITLE
[question] Whats the point on restoring purchases?

### DISCUSSION THREAD

**Discussion by nriccar** [hyochan/react-native-iap#2415] [Q&A] [answered]
Hello to all!

First of all, thanks for reading.

I have this doubt, it could seem naive from my part but I don't really get the point.
If a user subscribes to a auto-renewal subscription, it will get access to X part on my app. We save that information in our DB.

If the same user wants to access that subscription from another device, our app would ask for login, and after it, that subscription would be accessible.

What does it do the restore purchase action itself?

**Comment by hszouszcz** [comment]
Actually it's great question and I would like to hear some thoughts about this too. When such situation can take place if you store subscription data on BE?

**Comment by goguda** [comment, marked as answer]
> Hello to all!
> 
> First of all, thanks for reading.
> 
> I have this doubt, it could seem naive from my part but I don't really get the point. If a user subscribes to a auto-renewal subscription, it will get access to X part on my app. We save that information in our DB.
> 
> If the same user wants to access that subscription from another device, our app would ask for login, and after it, that subscription would be accessible.
> 
> What does it do the restore purchase action itself?

I remember back in the day, especially on iOS, a lot of apps would lose purchases if you deleted the app from your device since everything was stored locally. If you reinstalled, you could "restore purchases" which would essentially tell the app you've already paid for certain IAPs so it could store that info locally and grant access to the paid feature(s) again.

With modern server-side storage, there seems to be less of a need for this, but I suppose they've left the functionality there for people that still want to store everything locally on the user's device.

If you're storing the data server-side, you probably don't need to use the restore purchases function.

**Reply by nriccar** [reply]
Thanks for the detailed response! Yes, I agree. I do keep track of the purchases on my backend, and at the reviewing process of Apple they rejected requesting for this functionality to be in the binary. After I explained that we use backend with remote notifications, they approved the build without needing for a restore purchases functionality.

**Comment by andresesfm** [comment]
You should do a restore purchase along with asking them to login. In that way you get the latest information about the subscription that you can use to update your DB. The problem with keeping it on your DB is that you don't account for changes to the subscription status. i.e. cancellations or renewals.

**Reply by nriccar** [reply]
But you do with remote notifications, right? If the user cancels, or renews, etc., Apple will inform it to the backend via remote notifications and the user access-status will be changed
